### ISSUE TITLE
Warning: Cannot find reference 'loggers' in '__init__.py'

### ISSUE THREAD

**Issue by MathijsdeBoer** [Lightning-AI/lightning#17663]
### Bug description

When setting up a new training session, one can specify their preferred logger. Unfortunately, it seems that PyCharm throws a warning whenever you try to import the `pytorch_lightning.loggers` or `lightning.pytorch.logger` module.

This is not a blocking issue, nor a high priority bug, as the code will run just fine, but it is a slight annoyance to have a perpetual warning in my IDE.

### What version are you seeing the problem on?

v2.0

### How to reproduce the bug

```python
import lightning.pytorch as pl

from myproject import MyDataModule, MyModel

trainer = pl.Trainer(
    logger=pl.loggers.TensorBoardLogger("logs/", name="my-model")  # <- Warning is here
)

dm = MyDataModule()
model = MyModel()

trainer.fit(model, datamodule=dm)
```

### Error messages and logs

```bash
Cannot find reference 'loggers' in '__init__.py' 
```

### Environment

<details>
  <summary>Current environment</summary>

* CUDA:
        - GPU:
                - Quadro RTX 4000
        - available:         True
        - version:           11.8
* Lightning:
        - lightning-utilities: 0.8.0
        - pytorch-lightning: 2.0.2
        - torch:             2.0.0+cu118
        - torchmetrics:      0.11.4
* Packages:
        - aiohttp:           3.8.4
        - aiosignal:         1.3.1
        - async-timeout:     4.0.2
        - attrs:             23.1.0
        - black:             22.12.0
        - certifi:           2022.12.7
        - cfgv:              3.3.1
        - charset-normalizer: 3.1.0
        - click:             8.1.3
        - colorama:          0.4.6
        - distlib:           0.3.6
        - filelock:          3.12.0
        - frozenlist:        1.3.3
        - fsspec:            2023.4.0
        - identify:          2.5.22
        - idna:              3.4
        - jinja2:            3.1.2
        - lightning-utilities: 0.8.0
        - markupsafe:        2.1.2
        - mpmath:            1.3.0
        - multidict:         6.0.4
        - mypy-extensions:   1.0.0
        - networkx:          3.1
        - nodeenv:           1.7.0
        - numpy:             1.24.3
        - packaging:         23.1
        - pathspec:          0.11.1
        - pip:               22.3.1
        - platformdirs:      3.2.0
        - pre-commit:        3.2.2
        - pytorch-lightning: 2.0.2
        - pyyaml:            6.0
        - requests:          2.28.2
        - ruff:              0.0.261
        - setuptools:        65.5.0
        - sympy:             1.11.1
        - torch:             2.0.0+cu118
        - torchmetrics:      0.11.4
        - tqdm:              4.65.0
        - typing-extensions: 4.5.0
        - urllib3:           1.26.15
        - virtualenv:        20.22.0
        - yarl:              1.9.1
* System:
        - OS:                Windows
        - architecture:
                - 64bit
                - WindowsPE
        - processor:         Intel64 Family 6 Model 85 Stepping 4, GenuineIntel
        - python:            3.11.3
        - release:           10
        - version:           10.0.19045

</details>

**Comment by awaelchli** [comment]
This could be because `__all__` in `lightning/pytorch/__init__.py` only contains a subset of importable objects and then PyCharm sees this as missing. But I don't see anything wrong with the current approach, it is weird that PyCharm flags this. Suggestions welcome.
